### PR TITLE
Make partial_algorithm work with any number of arguments

### PR DESCRIPTION
**Pull request by msimberg**
I previously only specialized it for zero and one arguments, out of laziness. This should make it work with any number.

### CODE REVIEW

File: libs/parallelism/execution/include/hpx/execution/algorithms/detail/partial_algorithm.hpp [hunk 1]
```diff
@@ -6,37 +6,47 @@
...
 #include <type_traits>
 #include <utility>
 
 namespace hpx {
     namespace execution {
         namespace experimental {
             namespace detail {
-    template <typename Tag, typename... Ts>
-    struct partial_algorithm;
+    template <typename Tag, typename Is, typename... Ts>
```
Comment (hkaiser) [inline libs/parallelism/execution/include/hpx/execution/algorithms/detail/partial_algorithm.hpp:19]: Nitpick: would you mind renaming `Is` to `Pack` (or something similar), please?

**Review by hkaiser — DISMISSED** [review]
LGTM, very nice, thanks!

File: libs/parallelism/execution/include/hpx/execution/algorithms/detail/partial_algorithm.hpp (hunk 1 above)
Comment (msimberg) [reply libs/parallelism/execution/include/hpx/execution/algorithms/detail/partial_algorithm.hpp:19]: I don't mind, but I named it `Is` to follow the convention in the functional module (and e.g. `just.hpp`, but that has the same motivation as here). At least I associate plain `Pack` with a pack of arguments like `Ts` so I'd prefer to keep the "index"-ness in the name somehow. `IsPack`? `IndexPack`?

**Review by aurianer — APPROVED** [review]
LGTM thanks!

File: libs/parallelism/execution/include/hpx/execution/algorithms/detail/partial_algorithm.hpp (hunk 1 above)
Comment (hkaiser) [reply libs/parallelism/execution/include/hpx/execution/algorithms/detail/partial_algorithm.hpp:19]: I meant just changing the name of the parameter for the main template. I'd be absolutely fine with leaving `Is...` for where it refers to a sequence of indexes. But mind you - this is _really_ minor.

Comment (msimberg) [reply libs/parallelism/execution/include/hpx/execution/algorithms/detail/partial_algorithm.hpp]: Don't worry; details matter. I changed it to `IsPack`.

**Pull request merged**